netjsonconfig/utils.py

```diff
@@ -32,6 +32,29 @@ def merge_config(template, config, list_identifiers=None):
     return result
 
 
+def get_common_keys(list1, list2):  # noqa: C901
+    """
+    Returns a list of common keys which have same values.
+    """
+    key_value_set = set()
+    common_keys = []
+    for el in list1:
+        if isinstance(el, dict):
+            for key, value in el.items():
+                if isinstance(value, (dict, list)):
+                    continue
+                if key not in key_value_set:
+                    key_value_set.add((key, el[key]))
+    for el in list2:
+        if isinstance(el, dict):
+            for key, value in el.items():
+                if isinstance(value, (dict, list)):
+                    continue
+                if (key, value) in key_value_set:
+                    common_keys.append(key)
+    return common_keys
```

Comment:
I don't understand what this code does, it is not clear at all.
Sankalp, this library is the base of OpenWISP and cannot be modified lightly, every change must be super clear and very specific, this change looks VERY broad. What does it means **Returns a list of common keys which have same values**???
You need to be very specific of WHAT keys are returned and why this function is introduced!

Moreover, again the approach is not the correct one in my opinion:  the problem we have is that we have two templates having the same CA file and the validator complains because the CA file is duplicated, then we need to write code which deals with this issue, not other code which is not even clear what it does.

I think what we have to do is to write clean code which CLEARLY expresses that if a list element in the second template is already present in the first template (as is), the second element is not added to the first list.